solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1443,20 +1559,15 @@ public void refreshAndWatch() {
      */
     public void refreshAndWatch(EventType eventType) {
       try {
-        if (eventType == null || eventType == EventType.NodeChildrenChanged) {
-          refreshAndWatchChildren();
-          if (eventType == EventType.NodeChildrenChanged) {
-            // only per-replica states modified. return
-            return;
-          }
-        }
-
-        DocCollection newState = fetchCollectionState(coll, this);
-        collectionWatches.updateDocCollection(coll, newState);
-        synchronized (getUpdateLock()) {
-          constructState(Collections.singleton(coll));
+        StatefulCollectionWatch s = collectionWatches.statefulWatchesByCollectionName.get(coll);
+        if (s.persistentWatcher != null) return;
+        synchronized (s) {
+          if (s.persistentWatcher != null) return;
+          s.persistentWatcher =
+              zkClient.persistentRecursiveWatch(
+                  DocCollection.getCollectionPathRoot(coll), s::handleWatch);
         }
-
+        refreshAndWatchChildren();
```

Comment:
We probably do not need to watch the children here, since this watcher does nothing. As far as i understand the design, we install a persistent recursive watch that watch both the state.json and PRS entries (recommend move such handling logic out of StatefulCollectionWatch and into here).

And it seems we might also want to do a one-off fetch on all children (for optimization?) to construct the `PerReplicaStates`, if that's the case, we might want to just do a `zkClient.getChildren` without passing any watcher so it wouldn't register any extra watch with the ZK